lib/Transforms/IPO/StructFieldCacheAnalysis.cpp

```diff
@@ -8,9 +8,52 @@
 //
 //===------------------------------------------------------------------------===//
 //
-// This pass performs analysis on cache-aware structure field accesses based on
-// the following paper and reports recommendations on changes to make on the
-// source code to improve performance.
+// This file implements the main function of StructFieldCacheAnalysis pass. This
+// pass will analyze struct field access patterns in a given program and prints
+// recommendations on how to reorganize the structs to gain performance.
+//
+// The pass is implemented as following steps:
+// 0. Prerequites: BlockFrequencyInfo and BranchProbabilityInfo
+// 1. Perform IR analysis to obtain and record all memory accesses on which
+// field
+//    of all structs defined in the program. This step involves interaction with
+//    class StructFieldAccessManager that calls functions in class
+//    StructFieldAccessInfo to complete IR analysis.
+// 2. Apply filters to structs and gather function arguments info. This step
+// will
+//    filter out some structs due to safety or performance concernts to narrow
+//    down structs to be analyzed. It involves HotnessAnalyzer class.
+//    This step also summarizes function calls that takes field address as
+//    argument into function definitions to facilitate analysis.
+// 3. Build close proximity graph for every pair of fields of each struct. It
+// needs
+//    to interact with StructFieldAccessManager to retrieve info for each struct
+//    and create an object of CloseProximityBuilder to build CP relations.
+//    The CloseProximityBuilder needs to build FieldReferenceGraph object first
+//    to help build CP relations in steps.
+//
```

Comment:
Your comments say nothing about using the CPG info to generate the actual recommendations for altering struct field layouts.